ui/src/groups/Members.tsx

```diff
@@ -0,0 +1,374 @@
+        title="Ban Member"
+        message={`Are you sure you want to ban ${member}?`}
+        confirmText="Ban"
+        loading={loadingBan}
+        onConfirm={ban(member)}
+        open={showBanConfirm}
+        setOpen={setShowBanConfirm}
+      />
+    </>
+  );
+});
+
+export default function Members() {
+  const [toggleSearch, setToggleSearch] = useState(false);
+  const [search, setSearch] = useState('');
+  const [rawInput, setRawInput] = useState('');
+  const onUpdate = useRef(
+    debounce((value: string) => {
+      setSearch(value);
+    }, 150)
+  );
+  const flag = useRouteGroup();
+  const group = useGroup(flag, true);
+  const members = useMemo(() => {
+    if (!group) {
+      return [];
+    }
+    return Object.keys(group.fleet).filter((k) => {
+      if ('shut' in group.cordon) {
+        return (
+          !group.cordon.shut.ask.includes(k) &&
+          !group.cordon.shut.pending.includes(k)
+        );
+      }
+      if (group.fleet[k].sects.includes('admin')) {
+        return false;
+      }
+      return true;
+    });
+  }, [group]);
+
+  const admins = useMemo(() => {
+    if (!group) {
+      return [];
+    }
+    return Object.keys(group.fleet).filter((k) => {
+      if (group.fleet[k].sects.includes('admin')) {
+        return true;
+      }
+      return false;
+    });
+  }, [group]);
+
+  const results = useMemo(
+    () =>
+      fuzzy
+        .filter(search, members)
+        .sort((a, b) => {
+          const filter = deSig(search) || '';
+          const left = deSig(a.string)?.startsWith(filter)
+            ? a.score + 1
+            : a.score;
+          const right = deSig(b.string)?.startsWith(filter)
+            ? b.score + 1
+            : b.score;
+
+          return right - left;
+        })
+        .map((result) => members[result.index]),
+    [search, members]
+  );
+
+  const onChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    setRawInput(value);
+    onUpdate.current(value);
+  }, []);
+
+  return (
+    <div className="flex h-full w-full flex-col overflow-hidden bg-white">
+      <MobileHeader
+        title={
+          !toggleSearch ? (
+            'Members'
+          ) : (
+            <input
+              autoFocus
+              className="input"
+              placeholder="Filter Members"
+              value={rawInput}
+              onChange={onChange}
+            />
+          )
+        }
+        pathBack=".."
+        action={
+          <div className="flex h-12 flex-row items-center justify-end space-x-2">
+            <button onClick={() => setToggleSearch(!toggleSearch)}>
+              {toggleSearch ? (
+                <XIcon className="h-6 w-6 text-gray-800" />
+              ) : (
+                <MagnifyingGlassMobileNavIcon className="h-6 w-6 text-gray-800" />
+              )}
+            </button>
+          </div>
+        }
+      />
+      <div className="h-full overflow-auto px-4">
```

Comment:
Nice, looks good! We will need to virtualize this list, though - it takes ~40s to render all members of Tlon Local.

Reply:
will do, chef